contracts/proxies/CurrencyConvertor.sol

```diff
@@ -71,46 +79,35 @@ contract CurrencyConvertor {
     * @param  data             Trade parameters for the ExchangeWrapper.
     */
   function deposit(
-    ERC20 tokenFrom,
+    IERC20 tokenFrom,
     uint256 tokenFromAmount,
-    address exchangeWrapper,
+    I_ExchangeWrapper exchangeWrapper,
     uint256 starkKey,
     uint256 positionId,
     bytes calldata data
   )
     external
     returns (uint256)
   {
+    address self = address(this);
 
-    // Send fromToken to the ExchangeWrapper.
-    IERC20(tokenFrom).safeTransferFrom(
+    // Send fromToken to this contract.
+    tokenFrom.safeTransferFrom(
       msg.sender,
-      exchangeWrapper,
+      address(exchangeWrapper),
       tokenFromAmount
     );
 
-    address self = address(this);
-
     // Convert fromToken to toToken on the ExchangeWrapper.
-    I_ExchangeWrapper exchangeWrapperContract = I_ExchangeWrapper(exchangeWrapper);
-
-    uint256 tokenToAmount = exchangeWrapperContract.exchange(
+    uint256 tokenToAmount = exchangeWrapper.exchange(
         msg.sender,
         self,
         USDC_ADDRESS,
-        tokenFrom,
+        IERC20(tokenFrom),
         tokenFromAmount,
         data
     );
```

Comment:
I don't think I see the point of using the exchange wrapper here. I'm assuming it's because we want to be able to use generic exchanges under the hood, but the point of an aggregrator like 0x is that they do that for you (since they source liquidity from multiple exchanges). IMO this contract should just interface with 0x, and shouldn't be exchange-agnostic (since that's what 0x is meant to do).

The benefits of getting rid of the exchange wrapper is that we don't have to do an additional transfer of `makerToken` (USDC) from the exchange wrapper to the currency convertor contract. IMO we want this since users do not want to pay gas fees for two transfers, and would rather go through this process manually (without the extra transfer) if it was cheaper.

Reply:
Im not touching this Q because we still need to decide if
1. we want an exchange wrapper or this whole thing is always going to use 0x
2. its ok for the exchange wrapper to send directly to starkware or not